uncaught exceptions in junit5 tests

```
2020-06-11T17:23:53.1081906Z 15044 [pool-4-thread-1] ERROR o.j.j.e.e.JupiterEngineExecutionContext:148 - Caught exception while closing extension context: org.junit.jupiter.engine.descriptor.MethodExtensionContext@4daf8903
2020-06-11T17:23:53.1083395Z java.lang.IllegalArgumentException: timeout must be zero or greater
2020-06-11T17:23:53.1084815Z 	at org.osgi.test.common.service.ServiceConfiguration.<init>(ServiceConfiguration.java:63)
2020-06-11T17:23:53.1085938Z 	at org.osgi.test.junit5.service.ServiceExtension.lambda$getServiceUseConfiguration$1(ServiceExtension.java:146)
2020-06-11T17:23:53.1087435Z 	at org.junit.jupiter.engine.execution.ExtensionValuesStore.lambda$getOrComputeIfAbsent$0(ExtensionValuesStore.java:81)
2020-06-11T17:23:53.1088786Z 	at org.junit.jupiter.engine.execution.ExtensionValuesStore$MemoizingSupplier.get(ExtensionValuesStore.java:182)
2020-06-11T17:23:53.1090130Z 	at org.junit.jupiter.engine.execution.ExtensionValuesStore.closeAllStoredCloseableValues(ExtensionValuesStore.java:58)
2020-06-11T17:23:53.1091344Z 	at org.junit.jupiter.engine.descriptor.AbstractExtensionContext.close(AbstractExtensionContext.java:73)
2020-06-11T17:23:53.1092549Z 	at org.junit.jupiter.engine.execution.JupiterEngineExecutionContext.close(JupiterEngineExecutionContext.java:53)
2020-06-11T17:23:53.1093709Z 	at org.junit.jupiter.engine.descriptor.JupiterTestDescriptor.cleanUp(JupiterTestDescriptor.java:222)
2020-06-11T17:23:53.1095029Z 	at org.junit.jupiter.engine.descriptor.JupiterTestDescriptor.cleanUp(JupiterTestDescriptor.java:57)
2020-06-11T17:23:53.1096165Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$cleanUp$9(NodeTestTask.java:151)
2020-06-11T17:23:53.1097328Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-06-11T17:23:53.1098440Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.cleanUp(NodeTestTask.java:151)
2020-06-11T17:23:53.1099558Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:83)
2020-06-11T17:23:53.1100747Z 	at java.util.ArrayList.forEach(ArrayList.java:1257)
2020-06-11T17:23:53.1103464Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
2020-06-11T17:23:53.1104532Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
2020-06-11T17:23:53.1105816Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-06-11T17:23:53.1111818Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
2020-06-11T17:23:53.1118520Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
2020-06-11T17:23:53.1127813Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
2020-06-11T17:23:53.1135743Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-06-11T17:23:53.1144203Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
2020-06-11T17:23:53.1151598Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
2020-06-11T17:23:53.1155854Z 	at java.util.ArrayList.forEach(ArrayList.java:1257)
2020-06-11T17:23:53.1167340Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
2020-06-11T17:23:53.1189751Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
2020-06-11T17:23:53.1198800Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-06-11T17:23:53.1208018Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
2020-06-11T17:23:53.1214900Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
2020-06-11T17:23:53.1245641Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
2020-06-11T17:23:53.1255434Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-06-11T17:23:53.1263975Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
2020-06-11T17:23:53.1271678Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
2020-06-11T17:23:53.1283106Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
2020-06-11T17:23:53.1293874Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
2020-06-11T17:23:53.1302234Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
2020-06-11T17:23:53.1309781Z 	at org.junit.platform.testkit.engine.EngineTestKit.execute(EngineTestKit.java:189)
2020-06-11T17:23:53.1316787Z 	at org.junit.platform.testkit.engine.EngineTestKit.access$100(EngineTestKit.java:54)
2020-06-11T17:23:53.1325837Z 	at org.junit.platform.testkit.engine.EngineTestKit$Builder.execute(EngineTestKit.java:343)
2020-06-11T17:23:53.1335766Z 	at org.osgi.test.junit5.service.AbstractServiceExtensionTest.assertThatTest(AbstractServiceExtensionTest.java:126)
2020-06-11T17:23:53.1347349Z 	at org.osgi.test.junit5.service.AbstractServiceExtensionTest.lambda$futureAssertThatTest$0(AbstractServiceExtensionTest.java:113)
2020-06-11T17:23:53.1353191Z 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
2020-06-11T17:23:53.1362322Z 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
2020-06-11T17:23:53.1372502Z 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
2020-06-11T17:23:53.1380238Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2020-06-11T17:23:53.1387473Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2020-06-11T17:23:53.1391941Z 	at java.lang.Thread.run(Thread.java:748)
2020-06-11T17:23:53.1809645Z 15121 [pool-7-thread-1] ERROR o.j.j.e.e.JupiterEngineExecutionContext:148 - Caught exception while closing extension context: org.junit.jupiter.engine.descriptor.MethodExtensionContext@1df1ebcc
2020-06-11T17:23:53.1811365Z java.lang.IllegalArgumentException: cardinality must be zero or greater
2020-06-11T17:23:53.1812232Z 	at org.osgi.test.common.service.ServiceConfiguration.<init>(ServiceConfiguration.java:58)
2020-06-11T17:23:53.1814089Z 	at org.osgi.test.junit5.service.ServiceExtension.lambda$getServiceUseConfiguration$1(ServiceExtension.java:146)
2020-06-11T17:23:53.1815141Z 	at org.junit.jupiter.engine.execution.ExtensionValuesStore.lambda$getOrComputeIfAbsent$0(ExtensionValuesStore.java:81)
2020-06-11T17:23:53.1816131Z 	at org.junit.jupiter.engine.execution.ExtensionValuesStore$MemoizingSupplier.get(ExtensionValuesStore.java:182)
2020-06-11T17:23:53.1817124Z 	at org.junit.jupiter.engine.execution.ExtensionValuesStore.closeAllStoredCloseableValues(ExtensionValuesStore.java:58)
2020-06-11T17:23:53.1818093Z 	at org.junit.jupiter.engine.descriptor.AbstractExtensionContext.close(AbstractExtensionContext.java:73)
2020-06-11T17:23:53.1819375Z 	at org.junit.jupiter.engine.execution.JupiterEngineExecutionContext.close(JupiterEngineExecutionContext.java:53)
2020-06-11T17:23:53.1820249Z 	at org.junit.jupiter.engine.descriptor.JupiterTestDescriptor.cleanUp(JupiterTestDescriptor.java:222)
2020-06-11T17:23:53.1821131Z 	at org.junit.jupiter.engine.descriptor.JupiterTestDescriptor.cleanUp(JupiterTestDescriptor.java:57)
2020-06-11T17:23:53.1822009Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$cleanUp$9(NodeTestTask.java:151)
2020-06-11T17:23:53.1823249Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-06-11T17:23:53.1824170Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.cleanUp(NodeTestTask.java:151)
2020-06-11T17:23:53.1825251Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:83)
2020-06-11T17:23:53.1826044Z 	at java.util.ArrayList.forEach(ArrayList.java:1257)
2020-06-11T17:23:53.1827005Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
2020-06-11T17:23:53.1828039Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
2020-06-11T17:23:53.1829011Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-06-11T17:23:53.1829959Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
2020-06-11T17:23:53.1830833Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
2020-06-11T17:23:53.1831769Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
2020-06-11T17:23:53.1832719Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-06-11T17:23:53.1833830Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
2020-06-11T17:23:53.1834714Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
2020-06-11T17:23:53.1835486Z 	at java.util.ArrayList.forEach(ArrayList.java:1257)
2020-06-11T17:23:53.1836386Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
2020-06-11T17:23:53.1837396Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
2020-06-11T17:23:53.1838467Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-06-11T17:23:53.1839608Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
2020-06-11T17:23:53.1840460Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
2020-06-11T17:23:53.1841346Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
2020-06-11T17:23:53.1842536Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-06-11T17:23:53.1843405Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
2020-06-11T17:23:53.1844269Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
2020-06-11T17:23:53.1845262Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
2020-06-11T17:23:53.1846237Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
2020-06-11T17:23:53.1920709Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
2020-06-11T17:23:53.1921689Z 	at org.junit.platform.testkit.engine.EngineTestKit.execute(EngineTestKit.java:189)
2020-06-11T17:23:53.1922629Z 	at org.junit.platform.testkit.engine.EngineTestKit.access$100(EngineTestKit.java:54)
2020-06-11T17:23:53.1923821Z 	at org.junit.platform.testkit.engine.EngineTestKit$Builder.execute(EngineTestKit.java:343)
2020-06-11T17:23:53.1924729Z 	at org.osgi.test.junit5.service.AbstractServiceExtensionTest.assertThatTest(AbstractServiceExtensionTest.java:126)
2020-06-11T17:23:53.1925878Z 	at org.osgi.test.junit5.service.AbstractServiceExtensionTest.lambda$futureAssertThatTest$0(AbstractServiceExtensionTest.java:113)
2020-06-11T17:23:53.1946280Z 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
2020-06-11T17:23:53.1947273Z 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
2020-06-11T17:23:53.1948272Z 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
2020-06-11T17:23:53.1949132Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2020-06-11T17:23:53.1949960Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2020-06-11T17:23:53.1950679Z 	at java.lang.Thread.run(Thread.java:748)
2020-06-11T17:23:53.6420972Z 15587 [ool-11-thread-1] ERROR o.j.j.e.e.JupiterEngineExecutionContext:148 - Caught exception while closing extension context: org.junit.jupiter.engine.descriptor.MethodExtensionContext@7100b8f1
2020-06-11T17:23:53.6422731Z org.osgi.framework.InvalidSyntaxException: Filter ended abruptly: (&(objectClass=org.osgi.test.junit5.types.Foo)(foo=baz)
2020-06-11T17:23:53.6423893Z 	at org.eclipse.osgi.internal.framework.FilterImpl$Parser.parse(FilterImpl.java:1378)
2020-06-11T17:23:53.6424933Z 	at org.eclipse.osgi.internal.framework.FilterImpl.newInstance(FilterImpl.java:180)
2020-06-11T17:23:53.6425937Z 	at org.eclipse.osgi.internal.framework.FilterImpl.newInstance(FilterImpl.java:176)
2020-06-11T17:23:53.6427372Z 	at org.osgi.framework.FrameworkUtil.createFilter(FrameworkUtil.java:72)
2020-06-11T17:23:53.6428342Z 	at org.osgi.test.common.exceptions.FunctionWithException.lambda$orElseThrow$0(FunctionWithException.java:21)
2020-06-11T17:23:53.6429221Z 	at org.osgi.test.common.filter.Filters.format(Filters.java:43)
2020-06-11T17:23:53.6430093Z 	at org.osgi.test.common.service.ServiceConfiguration.<init>(ServiceConfiguration.java:53)
2020-06-11T17:23:53.6431460Z 	at org.osgi.test.junit5.service.ServiceExtension.lambda$getServiceUseConfiguration$1(ServiceExtension.java:146)
2020-06-11T17:23:53.6432777Z 	at org.junit.jupiter.engine.execution.ExtensionValuesStore.lambda$getOrComputeIfAbsent$0(ExtensionValuesStore.java:81)
2020-06-11T17:23:53.6433911Z 	at org.junit.jupiter.engine.execution.ExtensionValuesStore$MemoizingSupplier.get(ExtensionValuesStore.java:182)
2020-06-11T17:23:53.6435472Z 	at org.junit.jupiter.engine.execution.ExtensionValuesStore.closeAllStoredCloseableValues(ExtensionValuesStore.java:58)
2020-06-11T17:23:53.6436612Z 	at org.junit.jupiter.engine.descriptor.AbstractExtensionContext.close(AbstractExtensionContext.java:73)
2020-06-11T17:23:53.6437918Z 	at org.junit.jupiter.engine.execution.JupiterEngineExecutionContext.close(JupiterEngineExecutionContext.java:53)
2020-06-11T17:23:53.6439091Z 	at org.junit.jupiter.engine.descriptor.JupiterTestDescriptor.cleanUp(JupiterTestDescriptor.java:222)
2020-06-11T17:23:53.6440262Z 	at org.junit.jupiter.engine.descriptor.JupiterTestDescriptor.cleanUp(JupiterTestDescriptor.java:57)
2020-06-11T17:23:53.6441534Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$cleanUp$9(NodeTestTask.java:151)
2020-06-11T17:23:53.6442832Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-06-11T17:23:53.6443891Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.cleanUp(NodeTestTask.java:151)
2020-06-11T17:23:53.6445548Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:83)
2020-06-11T17:23:53.6446707Z 	at java.util.ArrayList.forEach(ArrayList.java:1257)
2020-06-11T17:23:53.6451087Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
2020-06-11T17:23:53.6454086Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
2020-06-11T17:23:53.6456329Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-06-11T17:23:53.6460064Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
2020-06-11T17:23:53.6461396Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
2020-06-11T17:23:53.6464189Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
2020-06-11T17:23:53.6466732Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-06-11T17:23:53.6469656Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
2020-06-11T17:23:53.6472310Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
2020-06-11T17:23:53.6473972Z 	at java.util.ArrayList.forEach(ArrayList.java:1257)
2020-06-11T17:23:53.6478322Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
2020-06-11T17:23:53.6481642Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
2020-06-11T17:23:53.6484175Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-06-11T17:23:53.6487086Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
2020-06-11T17:23:53.6489080Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
2020-06-11T17:23:53.6493654Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
2020-06-11T17:23:53.6527240Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-06-11T17:23:53.6529803Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
2020-06-11T17:23:53.6532661Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
2020-06-11T17:23:53.6536534Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
2020-06-11T17:23:53.6539352Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
2020-06-11T17:23:53.6542320Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
2020-06-11T17:23:53.6544245Z 	at org.junit.platform.testkit.engine.EngineTestKit.execute(EngineTestKit.java:189)
2020-06-11T17:23:53.6546908Z 	at org.junit.platform.testkit.engine.EngineTestKit.access$100(EngineTestKit.java:54)
2020-06-11T17:23:53.6550043Z 	at org.junit.platform.testkit.engine.EngineTestKit$Builder.execute(EngineTestKit.java:343)
2020-06-11T17:23:53.6552658Z 	at org.osgi.test.junit5.service.AbstractServiceExtensionTest.assertThatTest(AbstractServiceExtensionTest.java:126)
2020-06-11T17:23:53.6556662Z 	at org.osgi.test.junit5.service.AbstractServiceExtensionTest.lambda$futureAssertThatTest$0(AbstractServiceExtensionTest.java:113)
2020-06-11T17:23:53.6558905Z 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
2020-06-11T17:23:53.6561702Z 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
2020-06-11T17:23:53.6564965Z 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
2020-06-11T17:23:53.6567344Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2020-06-11T17:23:53.6569511Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2020-06-11T17:23:53.6571205Z 	at java.lang.Thread.run(Thread.java:748)
2020-06-11T17:23:53.6573020Z Caused by: java.lang.ArrayIndexOutOfBoundsException: 55
2020-06-11T17:23:53.6576495Z 	at org.eclipse.osgi.internal.framework.FilterImpl$Parser.parse_and(FilterImpl.java:1445)
2020-06-11T17:23:53.6579165Z 	at org.eclipse.osgi.internal.framework.FilterImpl$Parser.parse_filtercomp(FilterImpl.java:1420)
2020-06-11T17:23:53.6581901Z 	at org.eclipse.osgi.internal.framework.FilterImpl$Parser.parse_filter(FilterImpl.java:1397)
2020-06-11T17:23:53.6584567Z 	at org.eclipse.osgi.internal.framework.FilterImpl$Parser.parse(FilterImpl.java:1376)
2020-06-11T17:23:53.6585347Z 	... 52 common frames omitted
```

I am not sure if these are expected or an error. If expected, we should try to get them not to appear in the build output.

--- a/org.osgi.test.junit5/src/test/java/org/osgi/test/junit5/context/BundleContextExtensionTest.java
+++ b/org.osgi.test.junit5/src/test/java/org/osgi/test/junit5/context/BundleContextExtensionTest.java
@@ -43,6 +43,8 @@ import java.util.Map;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
@@ -109,23 +111,34 @@ public class BundleContextExtensionTest {
 
 			AtomicReference<TestDescriptor> root = new AtomicReference<>();
 
-			EngineTestKit.engine(new JupiterTestEngine())
-				.selectors(selectClass(testClass))
-				.execute()
-				.allEvents()
-				// .debug(
-				// System.err)
-				.stream()
-				.filter(event -> event.getType()
-					.equals(FINISHED))
-				.forEach(event -> {
-					TestDescriptor current = event.getTestDescriptor();
-					eventMap.put(current, event);
-					if (!current.getParent()
-						.isPresent()) {
-						root.set(current);
-					}
-				});
+			Logger logger = Logger.getLogger("org.junit.jupiter");
+			Level oldLevel = logger.getLevel();
+			try {
+				// Suppress log output while the testkit is running (see issue
+				// #133).
+				logger.setLevel(Level.OFF);
+				EngineTestKit.engine(new JupiterTestEngine())
+					.selectors(selectClass(testClass))
+					.execute()
+					.allEvents()
+					// .debug(
+					// System.err)
+					.stream()
+					.filter(event -> event.getType()
+						.equals(FINISHED))
+					.forEach(event -> {
+						TestDescriptor current = event.getTestDescriptor();
+						eventMap.put(current, event);
+						if (!current.getParent()
+							.isPresent()) {
+							root.set(current);
+						}
+					});
+			} finally {
+				// Restore the filter to what it was so that we do not interfere
+				// with the parent test
+				logger.setLevel(oldLevel);
+			}
 
 			OSGiSoftAssertions afterSoftly = new OSGiSoftAssertions();
 			checker.assertSetup(afterSoftly, AFTER_CLASS, expectedSetup);

--- a/org.osgi.test.junit5/src/test/java/org/osgi/test/junit5/service/AbstractServiceExtensionTest.java
+++ b/org.osgi.test.junit5/src/test/java/org/osgi/test/junit5/service/AbstractServiceExtensionTest.java
@@ -11,6 +11,8 @@ import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 import org.assertj.core.api.AbstractThrowableAssert;
 import org.junit.jupiter.api.AfterEach;
@@ -110,9 +112,8 @@ abstract class AbstractServiceExtensionTest {
 	protected AbstractThrowableAssert<?, ? extends Throwable> futureAssertThatTest(Class<?> testClass, int delay) {
 		checkClass(testClass);
 		try {
-			return executor.schedule(() -> assertThatTest(testClass), delay,
-				TimeUnit.MILLISECONDS)
-				.get(delay + 2000, TimeUnit.MILLISECONDS);
+			return executor.schedule(() -> assertThatTest(testClass), delay, TimeUnit.MILLISECONDS)
+				.get(delay + 200000, TimeUnit.MILLISECONDS);
 		} catch (InterruptedException | ExecutionException | TimeoutException e) {
 			throw Exceptions.duck(e);
 		}
@@ -120,24 +121,34 @@ abstract class AbstractServiceExtensionTest {
 
 	protected static AbstractThrowableAssert<?, ? extends Throwable> assertThatTest(Class<?> testClass) {
 		checkClass(testClass);
-		Event testEvent = EngineTestKit.engine(new JupiterTestEngine())
-			.selectors(selectClass(
-				testClass))
-			.execute()
-			.testEvents()
-			// .debug(
-			// System.err)
-			.filter(event -> event.getType()
-				.equals(
-					FINISHED))
-			.findAny()
-			.orElseThrow(() -> new IllegalStateException("Test failed to run at all"));
 
-		TestExecutionResult result = testEvent.getPayload(TestExecutionResult.class)
-			.orElseThrow(() -> new IllegalStateException("Test result payload missing"));
+		Logger logger = Logger.getLogger("org.junit.jupiter");
+		Level oldLevel = logger.getLevel();
+		try {
+			// Suppress log output while the testkit is running (see issue
+			// #133).
+			logger.setLevel(Level.OFF);
+			Event testEvent = EngineTestKit.engine(new JupiterTestEngine())
+				.selectors(selectClass(testClass))
+				.execute()
+				.testEvents()
+				// .debug(
+				// System.err)
+				.filter(event -> event.getType()
+					.equals(FINISHED))
+				.findAny()
+				.orElseThrow(() -> new IllegalStateException("Test failed to run at all"));
 
-		return assertThat(result.getThrowable()
-			.orElse(null));
+			TestExecutionResult result = testEvent.getPayload(TestExecutionResult.class)
+				.orElseThrow(() -> new IllegalStateException("Test result payload missing"));
+
+			return assertThat(result.getThrowable()
+				.orElse(null));
+		} finally {
+			// Restore the filter to what it was so that we do not interfere
+			// with the parent test
+			logger.setLevel(oldLevel);
+		}
 	}
 
 	protected ScheduledFuture<ServiceRegistration<Foo>> schedule(Foo afoo, String key, String value) {

